File: be-py/utily/check.py
#!/usr/bin/env python3


def check_code(code):
    code = str(code)
    length = len(code)
    check_mark = code[-1]
    c1 = 0
    c2 = 0
    c3 = 0
    i = length - 2
    while i >= 0:
        c1 += int(code[i])
        i -= 2
    i = length - 3
    while i >= 0:
        c1 += int(code[i])
        i -= 2
    c3 = 10 - (c1 * 3 + c2) % 10
    if c3 == 10:
        c3 = 0
    if c3 == check_mark:
        return -1
    else:
        return c3


def get_check_code(code):
    code = str(code)
    length = len(code)
    c1 = 0
    c2 = 0
    c3 = 0
    i = length - 1
    while i >= 0:
        c1 += int(code[i])
        i -= 2
    i = length - 2
    while i >= 0:
        c1 += int(code[i])
        i -= 2
    c3 = 10 - (c1 * 3 + c2) % 10
    if c3 == 10:
        c3 = 0
    return c3

File: be/utily/check.js
module.exports = {
  checkCode: function(code){
    code = code.toString();
    code = code.trim();
    length = code.length;
    checkMark = code.charAt(length-1);
    for(var i=length-2,c1=0;i>=0;i-=2){
      c1 += Number(code.charAt(i));
    }
    for(var i=length-3,c2=0;i>=0;i-=2){
      c2 += Number(code.charAt(i))
    }
    c3 = 10-(c1*3+c2)%10;
    if(c3==10){
      c3=0;
    }
    if(c3==checkMark){
      console.log("检验数正确");
      return -1;
    }else {
      console.log("检验数错误，正确检验数应为" + c3);
      return c3;
    }
  },
  getCheckCode: function(code){
    code = code.toString();
    code = code.trim();
    length = code.length;
    for(var i=length-1,c1=0;i>=0;i-=2){
      c1 += Number(code.charAt(i));
    }
    for(var i=length-2,c2=0;i>=0;i-=2){
      c2 += Number(code.charAt(i))
    }
    c3 = 10-(c1*3+c2)%10;
    if(c3==10){
      c3=0;
    }
    console.log("检验数:" + c3);
    return c3;
  }
}
